stats/memory.go

```diff
@@ -0,0 +1,116 @@
+// Licensed under the Apache License, Version 2.0
+// Details: https://raw.githubusercontent.com/maniksurtani/quotaservice/master/LICENSE
+
+package stats
+
+import (
+	"sort"
+
+	"github.com/maniksurtani/quotaservice/events"
+)
+
+type namespaceStats struct {
+	hits, misses map[string]*BucketScore
+}
+
+type memoryListener struct {
+	namespaces map[string]*namespaceStats
+}
+
+func NewMemoryStatsListener() *memoryListener {
+	return &memoryListener{make(map[string]*namespaceStats)}
+}
+
+func (l *memoryListener) bucketScoreTop10(scoreMap map[string]*BucketScore) []*BucketScore {
+	arr := make(BucketScoreArray, 0)
+
+	for _, value := range scoreMap {
+		arr = append(arr, value)
+	}
+
+	sort.Sort(arr)
+	length := len(arr)
+
+	if length > 10 {
+		length = 10
+	}
+
+	return arr[0:length]
+}
+
+func (l *memoryListener) TopHits(namespace string) []*BucketScore {
```

Comment:
Even here it's idiomatic to add a comment to the effect of `TopHits implements StatsListener.TopHits` or something to that effect.